azkaban-common/src/test/java/azkaban/executor/DispatcherTest.java

```diff
@@ -0,0 +1,635 @@
+          return result == 0 ? 0 : result > 0 ? 1 : -1;
+
+        } }));
+    }
+
+    public void registerComparerForRemainingSpace(int weight){
+      this.registerFactorComparator(FactorComparator.create("RemainingTmp", weight, new Comparator<MockExecutorObject>(){
+        public int compare(MockExecutorObject o1, MockExecutorObject o2) {
+          int result = 0 ;
+
+          // check remaining % .
+          result = (int)(o1.remainingTmp - o2.remainingTmp);
+          return result == 0 ? 0 : result > 0 ? 1 : -1;
+
+        } }));
+    }
+
+    public void registerComparerForPriority(int weight){
+      this.registerFactorComparator(FactorComparator.create("Priority", weight, new Comparator<MockExecutorObject>(){
+        public int compare(MockExecutorObject o1, MockExecutorObject o2) {
+          int result = 0 ;
+
+          // check priority, bigger the better.
+          result = (int)(o1.priority - o2.priority);
+          return result == 0 ? 0 : result > 0 ? 1 : -1;
+
+        } }));
+    }
+  }
+
+  // test samples.
+  protected ArrayList<MockExecutorObject> executorList = new ArrayList<MockExecutorObject>();
+
+  @Before
+  public void setUp() throws Exception {
+    BasicConfigurator.configure();
+
+    executorList.clear();
+    executorList.add(new MockExecutorObject("Executor1",8080,50.0,2048,5,new Date(), 20, 6400));
+    executorList.add(new MockExecutorObject("Executor2",8080,50.0,2048,4,new Date(), 20, 6400));
+    executorList.add(new MockExecutorObject("Executor3",8080,40.0,2048,1,new Date(), 20, 6400));
+    executorList.add(new MockExecutorObject("Executor4",8080,50.0,2048,4,new Date(), 20, 6400));
+    executorList.add(new MockExecutorObject("Executor5",8080,50.0,1024,5,new Date(), 90, 6400));
+    executorList.add(new MockExecutorObject("Executor6",8080,50.0,1024,5,new Date(), 90, 3200));
+    executorList.add(new MockExecutorObject("Executor7",8080,50.0,1024,5,new Date(), 90, 3200));
+    executorList.add(new MockExecutorObject("Executor8",8080,50.0,2048,1,new Date(), 90, 3200));
+    executorList.add(new MockExecutorObject("Executor9",8080,50.0,2050,5,new Date(), 90, 4200));
+    executorList.add(new MockExecutorObject("Executor10",8080,00.0,1024,1,new Date(), 90, 3200));
+    executorList.add(new MockExecutorObject("Executor11",8080,20.0,2096,3,new Date(), 90, 2400));
+    executorList.add(new MockExecutorObject("Executor12",8080,90.0,2050,5,new Date(), 60, 2500));
+
+
+    // make sure each time the order is different.
+    Collections.shuffle(this.executorList);
+  }
+
+  private MockExecutorObject  getExecutorByName(String name){
+    MockExecutorObject returnVal = null;
+    for (MockExecutorObject item : this.executorList){
+      if (item.name.equals(name)){
+        returnVal = item;
+        break;
+      }
+    }
+    return returnVal;
+  }
+
+  @After
+  public void tearDown() throws Exception {
+  }
+
+  @Test
+  public void testExecutorFilter() throws Exception {
```

Comment:
for this specific one I think it is fine coz the test just try out some cases against a single API, I will add boundary and negative tests in separate methods.